src/components/graph-sonification.tsx

```diff
@@ -0,0 +1,174 @@
+      }
+    });
+  };
+
+  useEffect(() => {
+    const fetchGraphData = async () => {
+      const res = await codapInterface.sendRequest({action: "get", resource: `component[${graphToSonify}]`}) as IResult;
+      const graphDetails = res.values;
+      if (!graphDetails) return;
+
+      if (currentGraphIdRef.current && currentGraphIdRef.current !== graphDetails.id) {
+        await removeROIAdornment(currentGraphIdRef.current);
+      }
+
+      currentGraphIdRef.current = graphDetails.id;
+
+      await codapInterface.sendRequest({
+        action: "create",
+        resource: `component[${graphDetails.id}].adornment`,
+        values: {
+          type: "Region of Interest",
+          primary: { "position": 0, "extent": 0.05 },
+          secondary: { "position": 0, "extent": "100%" }
+        }
+      });
+
+      const pitchAttr = graphDetails.yAttributeName;
+      const timeAttr = graphDetails.xAttributeName;
+
+      const allItemsRes = await getAllItems(graphDetails.dataContext);
+      const allItems = allItemsRes.values;
+
+      // Asign pitch and time values to the dataset items
+      // Do not include items that are missing values for either attributes
+      const validItems = pitchAttr && timeAttr
+        ? allItems.filter((item: CodapItem) => item.values[pitchAttr] !== "" && item.values[timeAttr] !== "")
+        : allItems.filter((item: CodapItem) => item.values[timeAttr] !== "");
+
+      const pitchValues = validItems.map((item: CodapItem) => item.values[pitchAttr]);
+      const timeValues = validItems.map((item: CodapItem) => item.values[timeAttr]);
+
+      const minTime = graphDetails.xLowerBound;
+      const maxTime = graphDetails.xUpperBound;
+      const timeRange = maxTime - minTime || 1;
+
+      const minPitch = graphDetails.yLowerBound;
+      const maxPitch = graphDetails.yUpperBound;
+      const pitchRange = maxPitch - minPitch || 1;
+
+      const timeFractions = timeValues.map((t: number) => (t - minTime) / timeRange);
+      const pitchFractions = pitchValues.map((p: number) => (p - minPitch) / pitchRange);
+
+      await Tone.start();
+      Tone.getTransport().cancel();
+      Tone.getTransport().stop();
+      Tone.getTransport().position = 0;
+
+      const totalDuration = 10;
+      const fractionGroups: Record<number, number[]> = {};
+      timeFractions.forEach((frac: number, i: number) => {
+        if (!fractionGroups[frac]) fractionGroups[frac] = [];
+        fractionGroups[frac].push(i);
+      });
+
+      const uniqueFractions = Object.keys(fractionGroups)
+        .map(parseFloat)
+        .sort((a, b) => a - b);
+
+      uniqueFractions.forEach((fraction) => {
+        const offsetSeconds = fraction * totalDuration;
+        const indices = fractionGroups[fraction];
+      
+        Tone.getTransport().scheduleOnce((time) => {
+          indices.forEach((i) => {
+            const pFrac = pitchFractions[i];
+            const freq = 220 + pFrac * (880 - 220);          
```

Comment:
This should be a function with the constants given meaningful names.

Reply:
We're doing something to address this in the next PR.